sql/src/main/java/io/crate/metadata/Schemas.java

```diff
@@ -106,6 +111,25 @@ public DocTableInfo getWritableTable(TableIdent tableIdent) {
             throw new UnsupportedOperationException(String.format(Locale.ENGLISH,
                 "%s is an alias. Write, Drop or Alter operations are not supported", tableInfo.ident()));
         }
+        if (docTableInfo.closed()) {
+            throw new UnsupportedOperationException(String.format(Locale.ENGLISH,
+                "The table %s is closed. Write, Drop or Alter operations (other than OPEN) are not supported",
+                tableInfo.ident()));
+        }
+        return docTableInfo;
+    }
+
+    public DocTableInfo getOpenableOrClosableTable(TableIdent tableIdent) {
+        TableInfo tableInfo = getTableInfo(tableIdent);
+        if (!(tableInfo instanceof DocTableInfo)) {
+            throw new UnsupportedOperationException(String.format(Locale.ENGLISH,
+                "The table %s is read-only. Write, Drop or Alter operations are not supported", tableInfo.ident()));
```

Comment:
@mfussenegger How would one handle aliases in this context? For example, aliases are not writable or dropable, _except_ when it is an orphaned alias, when it is then dropable. Would it make more sense in this case to move the orphaned alias logic up to the DropTableAnalyzer (which is the only place that uses getDroppableTable) and instead just have a `getModifiableTable` function that checks whether or not the table is a DocTable and whether it is closed.

Related to this, given what @andreidan said, would it make more sense to reword the 'read-only' error message to state that the table is an internal/system table, and thus read only? Or are there non internal/system tables that are also not doctables?